src/main/java/org/folio/service/email/EmailService.java

```diff
@@ -0,0 +1,132 @@
+    this.configurationRestClient = configurationRestClient;
+    this.userRestClient = userRestClient;
+    this.dbClientFactory = dbClientFactory;
+    this.ledgerDAO = ledgerDAO;
+  }
+
+  public void createAndSendEmail(RequestContext requestContext, LedgerFiscalYearRollover rollover) {
+    getHostAddress(requestContext)
+      .onSuccess(hostAddressResponse -> getCurrentUser(requestContext)
+        .onSuccess(userResponse -> {
+          DBClient client = dbClientFactory.getDbClient(requestContext);
+          ledgerDAO.getLedgerById(rollover.getLedgerId(), client)
+            .onSuccess(ledger -> {
+              String hostAddress = hostAddressResponse.getJsonArray(CONFIGS_KEY).getJsonObject(0).getMap().get(VALUE_KEY).toString();
+              String linkToRolloverLedger = createRolloverLedgerLink(hostAddress, rollover.getLedgerId());
+              Map<String, String> headers = getHeaders(requestContext);
+              EmailEntity emailEntity = getEmailEntity(rollover, linkToRolloverLedger, ledger.getName(), userResponse);
+
+              sentEmail(requestContext, headers, emailEntity);
+            }).onFailure(t -> logger.error("Getting ledger failed {}", t.getMessage()));
+        }).onFailure(t -> logger.error("Getting user failed {}", t.getMessage())))
+      .onFailure(t -> logger.error("Getting host address failed {}", t.getMessage()));
+  }
+
+  private void sentEmail(RequestContext requestContext, Map<String, String> headers, EmailEntity emailEntity) {
+    EmailOkapiClient emailOkapiClient = new EmailOkapiClient(requestContext.getHeaders().get(OKAPI_URL), requestContext.getContext().owner(), headers);
+    emailOkapiClient.sendEmail(EMAIL_ENDPOINT, JsonObject.mapFrom(emailEntity).toString());
+  }
+
+  private EmailEntity getEmailEntity(LedgerFiscalYearRollover rollover, String linkToRolloverLedger, String ledgerName, JsonObject user) {
+    String email = user.getJsonObject(PERSONAL_KEY).getString(EMAIL_KEY);
+    EmailEntity emailEntity = new EmailEntity();
+    emailEntity.setNotificationId(UUID.randomUUID().toString());
+    emailEntity.setBody(getRolloverBody(rollover, linkToRolloverLedger, ledgerName, user));
+    emailEntity.setHeader(EMAIL_HEADER);
+    emailEntity.setTo(email);
+    emailEntity.setOutputFormat(MediaType.TEXT_HTML);
+    return emailEntity;
+  }
+
+  private String getRolloverBody(LedgerFiscalYearRollover rollover, String linkToRolloverLedger, String ledgerName, JsonObject user) {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(FORMAT_PATTERN).withZone(ZoneId.of(UTC_ZONE_ID));
+    return String.format(RolloverType.COMMIT.equals(rollover.getRolloverType()) ? getCommitMessage() : getPreviewMessage(),
+      user.getString(USERNAME_KEY),
+      formatter.format(rollover.getMetadata().getCreatedDate().toInstant()),
+      ledgerName,
+      linkToRolloverLedger);
+  }
+
+  private String getPreviewMessage() {
```

Comment:
nitpick: `getPreviewMessage` and `getCommitMessage` might benefit from compiler optimizations if they are declared as constants.